### PR TITLE
Add simple pass to turn dense attributes into dense_resource attributes.

### PR DESCRIPTION
**Pull request by stellaraccident**

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        op->setAttrs(attrs);
+    });
+    LLVM_DEBUG(llvm::dbgs() << "DONE CONVERTING RESOURCES\n");
+  }
+
+  static bool shouldConvertElements(ElementsAttr attr) {
+    if (llvm::isa<DenseElementsAttr>(attr)) {
+      // DenseElementsAttr encodes arbitrary dimension
+      // splats whereas DenseResourceElementsAttr does not.
+      if (attr.isSplat())
```
Comment (jpienaar) [inline compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp]: Return !attr.isSplat(); ?

File: compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        return BlobResourceElementsAttr::get(st, "dense_elements_i8",
+                                             std::move(blob));
+      case 16:
+        blob = HeapAsmResourceBlob::allocate(2 * numElements, /*align=*/64,
+                                             /*dataIsMutable=*/true);
+        copyIntAttrIntoBlob<uint16_t>(blob, attr);
+        return BlobResourceElementsAttr::get(st, "dense_elements_i16",
+                                             std::move(blob));
+      case 32:
+        blob = HeapAsmResourceBlob::allocate(4 * numElements, /*align=*/64,
```
Comment (jpienaar) [inline compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp:151]: Wouldn't the majority here be bitWidth/8? (Joining supported cases)

File: compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+#include "iree/compiler/Dialect/Util/Transforms/Passes.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinDialect.h"
+#include "mlir/Pass/Pass.h"
+
+#define DEBUG_TYPE "iree-util-import-resources"
+
+namespace mlir {
```
Comment (jpienaar) [inline compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp]: C++17 nested namespaces can.be used now

**Review by jpienaar — APPROVED** [review]
Useful for staging and transition but +1 should be higher.

File: compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp (hunk 2 above)
Comment (stellaraccident) [reply compiler/src/iree/compiler/Dialect/Util/Transforms/ImportResources.cpp:151]: Yeah, but there's some template goo in the copy that is needed to appease DenseElementsAttr...

**Pull request merged**